autoload/ctrlp/mrufiles.vim

```diff
@@ -56,16 +56,17 @@ fu! s:reformat(mrufs, ...)
 endf
 
 fu! s:record(bufnr)
-	if s:exclnomod && &l:modifiable | retu | en
-  if s:exclnomod && !&l:modifiable | retu | en
 	if s:locked | retu | en
+
 	let bufnr = a:bufnr + 0
 	let bufname = bufname(bufnr)
-	if bufnr > 0 && !empty(bufname)
-		cal filter(s:mrbs, 'v:val != bufnr')
-		cal insert(s:mrbs, bufnr)
-		cal s:addtomrufs(bufname)
-	en
+
+	if bufnr == 0 || !filereadable(bufname) | retu | en
```

Comment:
Looks like the only real change, apart from some reordering, is that we're now checking for `filereadable` before recording the buffer as an MRU? Doesn't that break a whole bunch of cases, like opening a new file (not written to disk), switch to something else, and then try to switch back to it using the MRU search?